internal/compiler-interface/src/main/java/xsbti/compile/analysis/Stamp.java

```diff
@@ -25,7 +25,7 @@
      *
      * @return A valid string-based representation for logical equality, not referential equality.
      */
-    public int getValueId();
+    public long getValueId();
```

Comment:
I'll reiterate that if you're going to change this, changing it to use `Array[Byte]` or `ByteBuffer` or protobuf's `ByteString` or something else that has dynamic length would be strongly preferable to using a fixed width value. It allows you to change the implementation of the hash function without changing the interface (which is fine for a hash function, as it just causes misses). In the future when you need more than 64 bits, it would not be necessary to change the API again.